sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/configuration/AadOAuth2ClientConfiguration.java

```diff
@@ -126,14 +126,13 @@ OAuth2AuthorizedClientManager authorizedClientManager(ClientRegistrationReposito
     JwtBearerOAuth2AuthorizedClientProvider azureAdJwtBearerProvider(ObjectProvider<OAuth2ClientAuthenticationJwkResolver> resolvers) {
         JwtBearerOAuth2AuthorizedClientProvider provider = new JwtBearerOAuth2AuthorizedClientProvider();
         OAuth2ClientAuthenticationJwkResolver resolver = resolvers.getIfUnique();
+        AadJwtBearerGrantRequestEntityConverter jwtBearerConverter = new AadJwtBearerGrantRequestEntityConverter();
```

Comment:
why moving this out will solve this issue?

Reply:
this converter is an extension against `JwtBearerGrantRequestEntityConverter`, which is provided by default in Spring Security, the OBO flow requires an extra parameter to interact with Azure AD, so this converter will always be added in `JwtBearerOAuth2AuthorizedClientProvider`.